service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/AzureVmHelper.java

```diff
@@ -44,6 +49,73 @@ public static StepResult deleteVm(
     return StepResult.getStepResultSuccess();
   }
 
+  public static StepResult removeAllUserAssignedManagedIdentitiesFromVm(
+      AzureCloudContext azureCloudContext, ComputeManager computeManager, String vmName)
+      throws InterruptedException {
+    try {
+      Set<String> userAssignedMsiIds =
+          computeManager
+              .virtualMachines()
+              .getByResourceGroup(azureCloudContext.getAzureResourceGroupId(), vmName)
+              .userAssignedManagedServiceIdentityIds();
+
+      if (userAssignedMsiIds != null) {
+        userAssignedMsiIds.forEach(
+            (String userAssignedMsiId) -> {
+              computeManager
+                  .virtualMachines()
+                  .getByResourceGroup(azureCloudContext.getAzureResourceGroupId(), vmName)
+                  .update()
+                  .withoutUserAssignedManagedServiceIdentity(userAssignedMsiId);
+            });
+      }
+    } catch (ManagementException e) {
+      handleNotFound(e, vmName, azureCloudContext.getAzureResourceGroupId());
```

Comment:
These calls to `handleNotFound` don't compile for me -- looks like they need another String param.